src/pages/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react"
+import io from "socket.io-client"
+import { GameController } from "../views/controller/game-controller"
+import { InQueue } from "../views/controller/in-queue"
+import { JoinGame } from "../views/controller/join-game"
+import { StartGame } from "../views/controller/start-game"
+import { NotRunning } from "../views/controller/not-running"
+import { GameOver } from "../views/controller/game-over"
+import { views } from "../lib/views"
+import { commands } from "../lib/commands"
+import { ControllerState } from "../server"
+
+interface IndexState {
+  socket: typeof io.Socket | null
+  activeView: string
+  queueLength: number
+}
+
+export default class Index extends Component<{}, IndexState> {
```

Comment:
Could we call this whole thing controller instead of index? Might be a bit more readable.